test/nbrowser/VersionUpdateBanner.ts

```diff
@@ -0,0 +1,69 @@
+import { version as installedVersion } from "app/common/version";
+import * as testUtils from 'test/server/testUtils';
+import * as gu from 'test/nbrowser/gristUtils';
+import {server, setupTestSuite} from 'test/nbrowser/testUtils';
+import {FakeUpdateServer, startFakeUpdateServer} from 'test/server/customUtil';
+import {assert, driver} from 'mocha-webdriver';
+
+describe('VersionUpdateBanner', function() {
+  this.timeout(300000);
+  setupTestSuite();
+
+  let oldEnv: testUtils.EnvironmentSnapshot;
+  let session: gu.Session;
+  let fakeServer: FakeUpdateServer;
+
+  afterEach(() => gu.checkForErrors());
+
+  before(async function() {
+    oldEnv = new testUtils.EnvironmentSnapshot();
+    process.env.GRIST_ALLOW_AUTOMATIC_VERSION_CHECKING = 'true';
+    process.env.GRIST_TEST_IMMEDIATE_VERSION_CHECK = 'true';
+    process.env.GRIST_DEFAULT_EMAIL = gu.session().email;
+    fakeServer = await startFakeUpdateServer();
+    process.env.GRIST_TEST_VERSION_CHECK_URL = `${fakeServer.url()}/version`;
+    await server.restart(true);
+  });
+
+  after(async function() {
+    await fakeServer.close();
+    oldEnv.restore();
+    await server.restart(true);
+  });
+
+  it('should not be shown to non-managers', async () => {
+    session = await gu.session().user('user2').personalSite.login();
+    await session.loadDocMenu('/');
+    assert.equal((await driver.findAll('.test-version-update-banner-text')).length, 0);
+  });
+
+  it('should be shown to managers', async () => {
+    session = await gu.session().personalSite.login();
+    await session.loadDocMenu('/');
+    assert.equal(await driver.find('.test-version-update-banner-text').isDisplayed(), true);
+
+    // Should be dismissable
+    await driver.find('.test-banner-close').click();
+    assert.equal((await driver.findAll('.test-version-update-banner-text')).length, 0);
+
+    // Let's make sure it's still not there after being dismissed once
+    await driver.navigate().refresh();
+    assert.equal((await driver.findAll('.test-version-update-banner-text')).length, 0);
+
+    // Update the version, the banner should come back
+    fakeServer.bumpVersion();
+    await server.restart(true);
+    session = await gu.session().personalSite.login({freshAccount: true});
+    await session.loadDocMenu('/');
+    assert.equal(await driver.find('.test-version-update-banner-text').isDisplayed(), true);
+  });
+
+  it('should not be shown to managers when there is no newer version', async () => {
+    fakeServer.latestVersion = installedVersion;
```

Comment:
Can you explain why this works ok? You are saving some things to localStorage and there are comments in the code that say that we don't we assume version is always new, but here you go back. It makes me think about what value is actually stored in localstorage. I think the banner was dismissed for this version in the upper test (for the same user), isn't it right?

It would be good maybe, to clear local storege for this user before this test, to make sure nothing was dissmissed